### ISSUE TITLE
"Cannot read property" error - WebGL init fail

### ISSUE THREAD

**Issue by shifters67** [GitSquared/edex-ui#217]
Please see the attached Pic
![screenshot from 2018-11-20 12-50-53](https://user-images.githubusercontent.com/7412647/48774866-4d08cb80-ecc3-11e8-912a-5a64d473733c.png)
Themes also fail to load

**Comment by GitSquared** [comment]
Thanks for the report. It seems like there's an issue loading WebGL on your system.

Could you please go to https://get.webgl.org/ and tell me what the page shows?

If you see the spinning cube and the "Your browser supports WebGL" message, I'll need to know what browser you use and your GPU to investigate further.

**Comment by shifters67** [comment]
Hi,

Firefox 63.0 and it says browser supports WebGL.  I am using the default drivers for my AMD Radeon 7970

regards

Tom

**Comment by GitSquared** [comment]
Thanks Tom. Could you test [this release I just uploaded](https://github.com/GitSquared/edex-ui/releases/download/v1.0.0/eDEX-UI.Linux.x86_64.AppImage) and tell me if the problem still happens?

**Comment by GitSquared** [comment]
****. I'll get back to you, i need to dig deep for this one.

**Comment by MohanChandar** [comment]
I downloaded the latest release just a few minutes ago and ran into the same issue. My system too supports WebGL. 
System specs are Ubuntu 18.04 - Budgie desktop - AMD A6 processor - Radeon graphics 512 MB

**Comment by GitSquared** [comment]
I think this might have something to do with AMD cards. Unfortunately it's way too low-level for me to do anything about it directly. I'll try setting up a test app later this week and sending it to you guys to see if i can isolate the problem.

(This might get fixed when i make the switch to electron 3.x)

**Comment by insipx** [comment]
Hey same issue, on linux, also with an AMD GPU (RX580).

Other system specs:
`uname -r` 4.19.2-300.fc29.x86_64
Distro: Fedora 29 KDE 5
Browser: Firefox 63.03 with WebGL Support

**Comment by Kstheking** [comment]
Same issue 
OS: Ubuntu 18.10
`uname -r`4.18.0-11-generic
Browser: Firefox 63.0 with WebGL Support

**Comment by MattMacs** [comment]
Same issue
OS: Windows 10 x64
Browser: Firefox 63.0.3 with WebGL Support
NVidia Quadro P2000

**Comment by GitSquared** [comment]
@MattMacs So you have a laptop with a "discrete" quadro GPU?

**Comment by MattMacs** [comment]
yes

**Comment by gazag** [comment]
Got the same problem, ran it from source, it fails there: src/assets/vendor/encom-globe.js:29518, the _gl variable is null.

The log was saying `Renderer is Chrome 61.0.3163.100`.

I tried an electron upgrade:
`npm install --save-dev electron@latest`
(it took electron@3.0.10)

Then the log says `Renderer is Chrome 66.0.3359.181`, and it works!

I'm not into js, i have no clue of the impact of an electron upgrade but it seems to work properly.

**Comment by GitSquared** [comment]
@gazag I already figured out the _gl error, but nice one trying electron v3. I'm actually waiting on them to release 3.0.11 which should include a fix for another issue similar to this one but affecting another range of video cards. (https://github.com/electron/electron/pull/15781)

So conclusion of this issue is wait for the next electron release to be bundled into eDEX. Shouldn't take too long i think.

**Comment by GitSquared** [comment]
Update: upstream PR has been merged to their 3.x branch, should see the end of this soon.

**Comment by MasterGeekMX** [comment]
Same isue Here:
OS: KDE Neon (Linux) [it's a fork of Ubuntu]
Kernel: 4.15.0-42-generic
Hardware:
- AMD Ryzen 5 1400
- 8GB RAM
- Radeon R7 360

I also have a ThinkPad T420 with Fedora 29 on there and there is no issues.

**Comment by GitSquared** [comment]
@MasterGeekMX This issue is fixed in the upcoming 2.0 release.

**Comment by didavid0609** [comment]
The issue also occurs on my MacBook Air 2011, but newer devices (MacOS) seem not to be affected.
